### PR TITLE
Liskov Substitution Principle (LSP)

### PR DESCRIPTION
**Pull request by peter-gribanov**
Fix for #37, #46

Correct **Good** example of LSP and add one more example.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1199,7 +1199,10 @@ foreach ($rectangles as $rectangle) {
 }
 ```
 
-**Good:**
+**Not good:**
+
+You can separate client for differen shapes.
```
Comment (alexislefebvre) [inline README.md]: A `t` is missing here: differen**t**

File: README.md [hunk 2]
```diff
@@ -1199,7 +1199,10 @@ foreach ($rectangles as $rectangle) {
...
+But this is not a best solution becous the square is still must be a subtype of the rectangle.
```
Comment (alexislefebvre) [inline README.md]: I suggest this:

> But this is not the best solution because

**Comment by peter-gribanov** [comment]
@TomasVotruba What do you think about this example? Is it not too complicated?

I want to add two more examples later with a violation of LSP so that readers can see different methods of breach of contract.

**Comment by TomasVotruba** [comment]
I got lost in changes and the original example as well. I don't have more free time to check this.

@jupeter needs to review

**Comment by peter-gribanov** [comment]
@TomasVotruba You can see changes in [split mode](https://github.com/jupeter/clean-code-php/pull/66/files?diff=split). Perhaps this will be easier to read.

**Comment by TomasVotruba** [comment]
Still too complicated for me

I think it can be as simple as 15 lines of code (after few secs of Googling): https://www.tomdalling.com/blog/software-design/solid-class-design-the-liskov-substitution-principle/

**Comment by peter-gribanov** [comment]
I understood you. I will make an easier example.

**Pull request closed without merging**